src/transformer/mock/mock.ts

```diff
@@ -63,3 +67,15 @@ export function getMockForList(nodeToMock: ts.TypeNode, node: ts.CallExpression)
 
     return ts.createArrayLiteral(mockList);
 }
+
+export function storeRegisterMock(typeToMock: ts.TypeNode, node: ts.CallExpression): ts.Node {
+    if (!ts.isTypeReferenceNode(typeToMock)) {
+        Logger('RegisterMock').error('registerMock can be used only to mock type references.');
+        return ts.createEmptyStatement();
+    }
+
+    const factory: ts.FunctionExpression = node.arguments[0] as ts.FunctionExpression;
+    MockDefiner.instance.storeRegisterMockFor(TypescriptHelper.GetDeclarationFromNode((typeToMock as ts.TypeReferenceNode).typeName), factory);
```

Comment:
if you want to avoid the casting to ts.TypeReferenceNode you can invert the if statement

You can also returning only once at the end with if else (I honestly don't mind either way).

Up to you :)

Reply:
makes sense, I've inverted the if, removed the casting and removed duplication